src/ngx_http_lua_regex.c

```diff
@@ -1922,12 +1924,78 @@ ngx_http_lua_ngx_re_sub_helper(lua_State *L, unsigned global)
 }
 
 
+ngx_int_t
+ngx_http_lua_set_jit_stack_size(int size)
+{
+    ngx_http_lua_main_conf_t    *lmcf;
+
+    lmcf = ngx_http_cycle_get_module_main_conf(ngx_cycle,
+                                               ngx_http_lua_module);
+
+    if (lmcf->regex_cache_entries > 0) {
+        return -1;
+    }
+
+    if (size < NGX_LUA_RE_MIN_JIT_STACK_SIZE) {
+        size = NGX_LUA_RE_MIN_JIT_STACK_SIZE;
+    }
+
+    if (lmcf->jit_stack) {
+        pcre_jit_stack_free(lmcf->jit_stack);
+    }
+
+    lmcf->jit_stack = pcre_jit_stack_alloc(NGX_LUA_RE_MIN_JIT_STACK_SIZE,
+                                           size);
+
+    if (!lmcf->jit_stack) {
+        return -2;
+    }
+
+    return 0;
+}
+
+
+static int
+ngx_http_lua_ngx_re_opt(lua_State *L)
+{
+    const char   *option;
+    int           nargs;
+    int           value = 0;
```

Comment:
This initialization is unneeded since `value` is always assigned before being used here.